io/pci/pci_hotplug.py

```diff
@@ -171,13 +175,41 @@ def net_recovery_check():
                     return True
             return False
 
+        def nvme_recovery_check():
+            '''
+            Checks if the nvme adapter functionality like all namespaces are
+            up and running or not after adapter is recovered
+            '''
+            err_ns = []
+            current_namespaces = nvme.get_current_ns_ids(self.contr_name)
+            if current_namespaces == self.ns_list:
+                for ns_id in current_namespaces:
+                    status = nvme.get_ns_status(self.contr_name, ns_id)
+                    if not status[0] == 'live' and status[1] == 'optimized':
+                        err_ns.append(ns_id)
+            else:
+                self.log.info("following ns not back listing after hot_plug" %
+                              (self.ns_list - current_namespaces))
+                return False
+
+            if err_ns:
+                self.log.info(f"following namespaces not recovered ={err_ns}")
+                return False
+            return True
+
         if wait.wait_for(is_added, timeout=30):
             time.sleep(45)
-            if pci.get_pci_class_name(pci_addr) == 'net':
+            if self.adapter_type == 'net':
                 if wait.wait_for(net_recovery_check, timeout=30):
                     return True
                 return False
-            else:
-                if wait.wait_for(is_recovered, timeout=30):
+            elif self.adapter_type == 'nvme':
+                if wait.wait_for(nvme_recovery_check, timeout=30):
+                    return True
+                return False
+            elif self.adapter_type == 'fc_host':
+                if wait.wait_for(fc_recovery_check, timeout=30):
                     return True
+                return False
+            return True
         return False
```

Comment:
so many returns ?

Reply:
yes, they are for all the different adapters
since it is designed in the same manner, I don't want change the existing design. if needed then will takup in next PR

Reply:
ok